File: scripts/extract-pdf.js
// scripts/extract-pdf.js
// Extract text from SnapStation PDF and save to data/photoBoothContent.txt

const fs = require('fs');
const path = require('path');
const pdf = require('pdf-parse');

const pdfPath = path.resolve(__dirname, '..', 'SnapStation.in — Premium Wooden Photo Booths _ Pune.pdf');
const outputPath = path.resolve(__dirname, '..', 'data', 'photoBoothContent.txt');

fs.readFile(pdfPath, (err, data) => {
  if (err) {
    console.error('Error reading PDF:', err);
    process.exit(1);
  }
  pdf(data)
    .then(result => {
      fs.mkdirSync(path.dirname(outputPath), { recursive: true });
      fs.writeFileSync(outputPath, result.text, 'utf8');
      console.log('Extracted text written to', outputPath);
    })
    .catch(e => {
      console.error('PDF parsing error:', e);
      process.exit(1);
    });
});

File: scripts/extract_pdf.py
#!/usr/bin/env python3
import pathlib
from pdfminer.high_level import extract_text

# Resolve paths relative to this script
pdf_path = pathlib.Path(__file__).resolve().parent.parent / "SnapStation.in — Premium Wooden Photo Booths _ Pune.pdf"
output_path = pathlib.Path(__file__).resolve().parent.parent / "data" / "photoBoothContent.txt"

# Extract text from PDF
text = extract_text(str(pdf_path))

# Ensure output directory exists
output_path.parent.mkdir(parents=True, exist_ok=True)

# Write extracted text
output_path.write_text(text, encoding="utf-8")
print(f"Extracted text written to {output_path}")
